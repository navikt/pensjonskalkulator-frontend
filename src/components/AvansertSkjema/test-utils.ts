export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
