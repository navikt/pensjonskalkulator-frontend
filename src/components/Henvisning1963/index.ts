export { Henvisning1963 as default } from './Henvisning1963'
