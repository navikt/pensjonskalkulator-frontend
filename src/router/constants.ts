export const BASE_PATH = '/pensjon/kalkulator'

export const externalUrls = {
  dinPensjon: 'https://nav.no/pensjon',
  dinPensjonBeholdning: 'https://www.nav.no/pensjon/opptjening/nb/',
  detaljertKalkulator: 'https://www.nav.no/pselv/simulering.jsf',
  alderspensjonsregler: 'https://www.nav.no/alderspensjon#beregning',
  afp: 'https://www.afp.no',
  afpPrivat: 'https://www.nav.no/afp-i-privat-sektor',
  garantipensjon: 'https://www.nav.no/minstepensjon',
  norskPensjon: 'https://norskpensjon.no/',
  uinnloggetKalkulator:
    'https://www.nav.no/pselv/simulering/forenkletsimulering.jsf',
  personvernerklaering:
    'https://www.nav.no/personvernerklaering#dine-rettigheter',
  personvernerklaeringKontaktOss:
    'https://www.nav.no/personvernerklaering#kontakt-nav',
  kontaktOss: 'https://www.nav.no/kontaktoss',
  planleggePensjon:
    'https://www.nav.no/planlegger-pensjon#noe-du-ikke-finner-svaret-p-her',
}

export const paths = {
  root: '/',
  login: '/login',
  henvisning: '/henvisning',
  start: '/start',
  samtykke: '/samtykke',
  utenlandsopphold: '/utenlandsopphold',
  offentligTp: '/offentlig-tp',
  afp: '/afp',
  samtykkeOffentligAFP: '/samtykke-offentlig-afp',
  ufoeretrygdAFP: '/ufoeretrygd-afp',
  sivilstand: '/sivilstand',
  uventetFeil: '/uventet-feil',
  beregningEnkel: '/beregning',
  beregningAvansert: '/beregning-detaljert',
  forbehold: '/forbehold',
  personopplysninger: '/personopplysninger',
} as const

export const henvisningUrlParams = {
  foedselsdato: '1963',
  apotekerne: 'apotekerne',
  utland: 'utland',
} as const
