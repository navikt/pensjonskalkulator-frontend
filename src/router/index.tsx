export { routes } from './routes'
export { externalUrls, paths, BASE_PATH } from './constants'
