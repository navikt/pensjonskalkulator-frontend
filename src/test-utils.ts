import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
