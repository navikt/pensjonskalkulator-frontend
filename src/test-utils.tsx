import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { getTranslation_test } from '@/utils/__tests__/test-translations'

import { createUttaksalderListener } from './state/listeners/uttaksalderListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
  hasRouter?: boolean
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations: Record<string, string> = getTranslation_nb()
  const testTranslations: Record<string, string> = getTranslation_test()
  const translationsInput = { ...nbTranslations, ...testTranslations }
  const translations: Record<string, string> = {}

  for (const key in translationsInput) {
    if (
      /<(?=.*? .*?\/ ?>|br|hr|input|!--|wbr)[a-z]+.*?>|<([a-z]+).*?<\/\1>|{[^}]+}/i.test(
        translationsInput[key]
      )
    ) {
      // for html or chunks keys: results in 'my_key' : 'my_key with some <html> or {chunk}'
      translations[key] = translationsInput[key]
    } else {
      // results in 'my_key' : 'my_key'
      translations[key] = key
    }
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    hasRouter = true,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale="nb" messages={generateMockedTranslations()}>
          {hasRouter ? <MemoryRouter>{children}</MemoryRouter> : children}
        </IntlProvider>
      </Provider>
    )
  }

  const listenerMiddleware = createListenerMiddleware()
  createUttaksalderListener(
    listenerMiddleware.startListening as AppStartListening
  )

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
