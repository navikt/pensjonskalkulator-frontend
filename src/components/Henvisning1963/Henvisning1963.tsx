import { FormattedMessage } from 'react-intl'

import { BodyLong, Button, Heading, HStack, VStack } from '@navikt/ds-react'

import { Card } from '../common/Card'
import { externalUrls } from '@/router'
import { wrapLogger } from '@/utils/logging'

interface IProps {
  onAvbryt: () => void
}

const gaaTilDetaljertKalkulator = () => {
  window.open(externalUrls.detaljertKalkulator, '_self')
}

export const Henvisning1963: React.FC<IProps> = ({ onAvbryt }) => {
  return (
    <Card
      data-testId="henvisning-1963"
      aria-live="polite"
      hasLargePadding
      hasMargin
    >
      <Heading level="2" size="medium" spacing>
        <FormattedMessage id="error.du_kan_ikke_bruke_enkel_kalkulator" />
      </Heading>
      <VStack gap="4">
        <BodyLong size="large">
          <FormattedMessage id="henvisning1963.body" />
        </BodyLong>

        <HStack gap="4">
          <Button
            type="button"
            data-testid="henvisning-1963.gaa-til-detaljert-kalkulator-knapp"
            variant="primary"
            onClick={wrapLogger('button klikk', { tekst: 'Tilbake' })(
              gaaTilDetaljertKalkulator
            )}
          >
            <FormattedMessage id="henvisning1963.detaljert_kalkulator" />
          </Button>
          <Button
            data-testid="henvisning-1963.avbryt-knapp"
            type="button"
            variant="secondary"
            onClick={wrapLogger('button klikk', { tekst: 'Avbryt' })(onAvbryt)}
          >
            <FormattedMessage id="stegvisning.avbryt" />
          </Button>
        </HStack>
      </VStack>
    </Card>
  )
}
