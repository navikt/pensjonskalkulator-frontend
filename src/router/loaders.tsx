import { redirect } from 'react-router'
import { defer, LoaderFunctionArgs, useLoaderData } from 'react-router-dom'

import { HOST_BASEURL } from '@/paths'
import { externalUrls, henvisningUrlParams, paths } from '@/router/constants'
import { apiSlice } from '@/state/api/apiSlice'
import { store } from '@/state/store'
import { selectIsVeileder, selectAfp } from '@/state/userInput/selectors'
import { isFoedtFoer1963 } from '@/utils/alder'
import { logger } from '@/utils/logging'
import { checkHarSamboer } from '@/utils/sivilstand'

export interface LoginContext {
  isLoggedIn: boolean
}

export function useDeferAuthenticationAccessData<
  TReturnedValue extends ReturnType<typeof authenticationDeferredLoader>,
>() {
  return useLoaderData() as ReturnType<TReturnedValue>['data']
}

{
  /* c8 ignore next 11 - Dette er kun for typing */
}
export function authenticationDeferredLoader<
  TData extends {
    oauth2Query: Response
  },
>(dataFunc: (args?: LoaderFunctionArgs) => TData) {
  return (args?: LoaderFunctionArgs) =>
    defer(dataFunc(args)) as Omit<ReturnType<typeof defer>, 'data'> & {
      data: TData
    }
}

export const authenticationGuard = async () => {
  const res = fetch(`${HOST_BASEURL}/oauth2/session`)
  return defer({
    oauth2Query: res,
  })
}

export const directAccessGuard = async () => {
  // Dersom ingen kall er registrert i store betyr det at brukeren prøver å aksessere en url direkte
  if (
    store.getState().api?.queries === undefined ||
    Object.keys(store.getState().api.queries).length === 0
  ) {
    return redirect(paths.start)
  }
  return null
}

// ////////////////////////////////////////

export function useLandingPageAccessData<
  TReturnedValue extends ReturnType<typeof landingPageDeferredLoader>,
>() {
  return useLoaderData() as ReturnType<TReturnedValue>['data']
}

{
  /* c8 ignore next 11 - Dette er kun for typing */
}
export function landingPageDeferredLoader<
  TData extends {
    shouldRedirectTo: string | undefined
  },
>(dataFunc: (args?: LoaderFunctionArgs) => TData) {
  return (args?: LoaderFunctionArgs) =>
    defer(dataFunc(args)) as Omit<ReturnType<typeof defer>, 'data'> & {
      data: TData
    }
}

export const landingPageAccessGuard = async () => {
  let resolveRedirectUrl: (
    value: string | PromiseLike<string>
  ) => void = () => {}

  const shouldRedirectTo: Promise<string> = new Promise((resolve) => {
    resolveRedirectUrl = resolve
  })

  const getPersonQuery = store.dispatch(apiSlice.endpoints.getPerson.initiate())
  getPersonQuery
    .then((res) => {
      if (
        res?.isSuccess &&
        isFoedtFoer1963(res?.data?.foedselsdato as string)
      ) {
        resolveRedirectUrl('')
        window.open(externalUrls.detaljertKalkulator, '_self')
      } else {
        if (selectIsVeileder(store.getState())) {
          resolveRedirectUrl(paths.start)
        } else {
          resolveRedirectUrl('')
        }
      }
    })
    .catch(() => {
      resolveRedirectUrl('')
    })

  return defer({
    shouldRedirectTo,
  })
}

/// ////////////////////////////////////////////////////////////////////////

export function useStep0AccessData<
  TReturnedValue extends ReturnType<typeof step0DeferredLoader>,
>() {
  return useLoaderData() as ReturnType<TReturnedValue>['data']
}

{
  /* c8 ignore next 11 - Dette er kun for typing */
}
export function step0DeferredLoader<
  TData extends {
    getPersonQuery: GetPersonQuery
    shouldRedirectTo: string | undefined
  },
>(dataFunc: (args?: LoaderFunctionArgs) => TData) {
  return (args?: LoaderFunctionArgs) =>
    defer(dataFunc(args)) as Omit<ReturnType<typeof defer>, 'data'> & {
      data: TData
    }
}

export const step0AccessGuard = async () => {
  let resolveRedirectUrl: (value: string | PromiseLike<string>) => void

  const shouldRedirectTo: Promise<string> = new Promise((resolve) => {
    resolveRedirectUrl = resolve
  })

  const getPersonQuery = store.dispatch(apiSlice.endpoints.getPerson.initiate())
  getPersonQuery.then((res) => {
    if (res?.isSuccess && isFoedtFoer1963(res?.data?.foedselsdato as string)) {
      window.open(externalUrls.detaljertKalkulator, '_self')
    }
  })

  // Henter inntekt til senere
  store.dispatch(apiSlice.endpoints.getInntekt.initiate())

  // Sørger for at brukeren er redirigert til henvisningsside iht. ekskludertStatus
  const getEkskludertStatusQuery = store.dispatch(
    apiSlice.endpoints.getEkskludertStatus.initiate()
  )

  getEkskludertStatusQuery.then((res) => {
    if (res?.data?.ekskludert) {
      if (res?.data?.aarsak === 'ER_APOTEKER') {
        resolveRedirectUrl(
          `${paths.henvisning}/${henvisningUrlParams.apotekerne}`
        )
      } else {
        resolveRedirectUrl('')
      }
    } else {
      resolveRedirectUrl('')
    }
  })

  return defer({
    getPersonQuery,
    shouldRedirectTo,
  })
}

// ///////////////////////////////////////////

export function useStep3AccessData<
  TReturnedValue extends ReturnType<typeof step3DeferredLoader>,
>() {
  return useLoaderData() as ReturnType<TReturnedValue>['data']
}

{
  /* c8 ignore next 11 - Dette er kun for typing */
}
export function step3DeferredLoader<
  TData extends {
    shouldRedirectTo: string | undefined
  },
>(dataFunc: (args?: LoaderFunctionArgs) => TData) {
  return (args?: LoaderFunctionArgs) =>
    defer(dataFunc(args)) as Omit<ReturnType<typeof defer>, 'data'> & {
      data: TData
    }
}

export const step3AccessGuard = async () => {
  if (await directAccessGuard()) {
    return redirect(paths.start)
  }
  const harSamtykket = store.getState().userInput.samtykke
  let resolveRedirectUrl: (value: string | PromiseLike<string>) => void
  let rejectRedirectUrl: (reason?: unknown) => void

  const shouldRedirectTo: Promise<string> = new Promise((resolve, reject) => {
    resolveRedirectUrl = resolve
    rejectRedirectUrl = reject
  })

  // Dersom brukeren prøver å aksessere steget direkte uten å ha svart på samtykke spørsmålet sendes den til start steget
  if (harSamtykket === null) {
    return redirect(paths.start)
  }

  // Dersom brukeren samtykker kaller vi tp-registret
  if (harSamtykket) {
    const getTpoMedlemskapQuery = store.dispatch(
      apiSlice.endpoints.getTpoMedlemskap.initiate()
    )
    // Dersom brukeren ikke har noe tp-forhold behøver ikke dette steget å vises
    getTpoMedlemskapQuery
      .then((res) => {
        if (res.isError) {
          throw new Error()
        }
        if (res?.isSuccess && !res?.data?.harTjenestepensjonsforhold) {
          resolveRedirectUrl(paths.afp)
        } else {
          resolveRedirectUrl('')
        }
      })
      .catch(() => {
        rejectRedirectUrl(null)
      })
    return defer({
      shouldRedirectTo,
    })
  } else {
    // Dersom brukeren ikke samtykker til henting av tpo behøver ikke dette steget å vises
    return redirect(paths.afp)
  }
}

/// ////////////////////////////////////////////////////////////////////////

export function useStep4AccessData<
  TReturnedValue extends ReturnType<typeof step4DeferredLoader>,
>() {
  return useLoaderData() as ReturnType<TReturnedValue>['data']
}

{
  /* c8 ignore next 11 - Dette er kun for typing */
}
export function step4DeferredLoader<
  TData extends {
    shouldRedirectTo: string | undefined
  },
>(dataFunc: (args?: LoaderFunctionArgs) => TData) {
  return (args?: LoaderFunctionArgs) =>
    defer(dataFunc(args)) as Omit<ReturnType<typeof defer>, 'data'> & {
      data: TData
    }
}

export const step4AccessGuard = async () => {
  if (await directAccessGuard()) {
    return redirect(paths.start)
  }
  let resolveRedirectUrl: (
    value: string | PromiseLike<string>
  ) => void = () => {}

  const shouldRedirectTo: Promise<string> = new Promise((resolve) => {
    resolveRedirectUrl = resolve
  })

  const hasInntektPreviouslyFailed = apiSlice.endpoints.getInntekt.select(
    undefined
  )(store.getState()).isError

  const hasEkskludertStatusPreviouslyFailed =
    apiSlice.endpoints.getEkskludertStatus.select(undefined)(
      store.getState()
    ).isError

  store.dispatch(apiSlice.endpoints.getUfoeregrad.initiate()).then((res) => {
    if (res.isError) {
      resolveRedirectUrl(paths.uventetFeil)
    }
    if (res.isSuccess) {
      // Hvis alle kallene er vellykket, resolve
      if (!hasInntektPreviouslyFailed && !hasEkskludertStatusPreviouslyFailed) {
        resolveRedirectUrl('')
      }
      // Hvis inntekt har feilet tidligere, prøv igjen og redirect til uventet feil ved ny feil
      if (hasInntektPreviouslyFailed) {
        store
          .dispatch(apiSlice.endpoints.getInntekt.initiate())
          .then((inntektRes) => {
            if (inntektRes.isError) {
              resolveRedirectUrl(paths.uventetFeil)
            } else if (
              apiSlice.endpoints.getEkskludertStatus.select(undefined)(
                store.getState()
              ).isSuccess
            ) {
              resolveRedirectUrl('')
            }
          })
      }
      // Hvis ekskludertStatus har feilet tidligere, prøv igjen og redirect til uventet feil ved ny feil
      if (hasEkskludertStatusPreviouslyFailed) {
        store
          .dispatch(apiSlice.endpoints.getEkskludertStatus.initiate())
          .then((ekskludertStatusRes) => {
            if (ekskludertStatusRes.isError) {
              logger('info', {
                tekst: 'ekskludert feilet',
              })
              resolveRedirectUrl(paths.uventetFeil)
            }
            if (res.isSuccess) {
              if (
                ekskludertStatusRes?.data?.ekskludert &&
                ekskludertStatusRes?.data?.aarsak === 'ER_APOTEKER'
              ) {
                resolveRedirectUrl(
                  `${paths.henvisning}/${henvisningUrlParams.apotekerne}`
                )
              } else if (
                apiSlice.endpoints.getInntekt.select(undefined)(
                  store.getState()
                ).isSuccess
              ) {
                resolveRedirectUrl('')
              }
            }
          })
      }
    }
  })

  return defer({
    shouldRedirectTo,
  })
}

/// ////////////////////////////////////////////////////////////////////////

export const step5AccessGuard = async () => {
  if (await directAccessGuard()) {
    return redirect(paths.start)
  }

  const afp = selectAfp(store.getState())
  const ufoereGradResponse = apiSlice.endpoints.getUfoeregrad.select(undefined)(
    store.getState()
  ).data

  if (ufoereGradResponse?.ufoeregrad && afp !== 'nei') {
    return null
  }
  return redirect(paths.sivilstand)
}

// ////////////////////////////////////////

export function useStep6AccessData<
  TReturnedValue extends ReturnType<typeof step6DeferredLoader>,
>() {
  return useLoaderData() as ReturnType<TReturnedValue>['data']
}

{
  /* c8 ignore next 11 - Dette er kun for typing */
}
export function step6DeferredLoader<
  TData extends {
    getPersonQuery: GetPersonQuery
    shouldRedirectTo: string | undefined
  },
>(dataFunc: (args?: LoaderFunctionArgs) => TData) {
  return (args?: LoaderFunctionArgs) =>
    defer(dataFunc(args)) as Omit<ReturnType<typeof defer>, 'data'> & {
      data: TData
    }
}

export const step6AccessGuard = async () => {
  if (await directAccessGuard()) {
    return redirect(paths.start)
  }
  let resolveRedirectUrl: (
    value: string | PromiseLike<string>
  ) => void = () => {}

  const shouldRedirectTo: Promise<string> = new Promise((resolve) => {
    resolveRedirectUrl = resolve
  })

  let resolveGetPerson: (
    value: null | GetPersonQuery | PromiseLike<GetPersonQuery>
  ) => void = () => {}

  const getPersonQuery: Promise<null | GetPersonQuery> = new Promise(
    (resolve) => {
      resolveGetPerson = resolve
    }
  )

  const getPersonPreviousResponse = apiSlice.endpoints.getPerson.select(
    undefined
  )(store.getState())

  // Hvis getPerson har blitt fetchet tidligere, gjenbruk responsen
  if (getPersonPreviousResponse.isSuccess) {
    if (
      getPersonPreviousResponse?.data?.sivilstand &&
      checkHarSamboer(getPersonPreviousResponse.data.sivilstand)
    ) {
      resolveRedirectUrl(paths.beregningEnkel)
      resolveGetPerson(getPersonPreviousResponse)
    } else {
      resolveRedirectUrl('')
      resolveGetPerson(getPersonPreviousResponse)
    }
  } else {
    // Hvis getPerson har feilet tidligere, prøv igjen og redirect til uventet feil ved ny feil
    const newGetPersonQuery = store.dispatch(
      apiSlice.endpoints.getPerson.initiate()
    )
    newGetPersonQuery.then((res) => {
      if (res.isError) {
        resolveRedirectUrl(paths.uventetFeil)
        resolveGetPerson(res)
      }
      if (res.isSuccess) {
        if (isFoedtFoer1963(res?.data?.foedselsdato as string)) {
          window.open(externalUrls.detaljertKalkulator, '_self')
          resolveRedirectUrl('')
          resolveGetPerson(res)
        } else if (
          res?.data?.sivilstand &&
          checkHarSamboer(res.data.sivilstand)
        ) {
          resolveRedirectUrl(paths.beregningEnkel)
          resolveGetPerson(res)
        } else {
          resolveRedirectUrl('')
          resolveGetPerson(res)
        }
      }
    })
  }

  return defer({
    getPersonQuery,
    shouldRedirectTo,
  })
}
