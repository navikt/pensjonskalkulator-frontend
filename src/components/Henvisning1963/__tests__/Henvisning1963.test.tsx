import { describe, it, vi } from 'vitest'

import { Henvisning1963 } from '../Henvisning1963'
import { externalUrls } from '@/router'
import { render, screen, userEvent } from '@/test-utils'

describe('Henvisning1963', async () => {
  it('rendrer', () => {
    const onAvbryt = vi.fn()
    const { asFragment } = render(<Henvisning1963 onAvbryt={onAvbryt} />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByTestId('henvisning-1963')).toBeVisible()
  })

  it('trykker avbryt knapp', async () => {
    const user = userEvent.setup()
    const onAvbryt = vi.fn()
    render(<Henvisning1963 onAvbryt={onAvbryt} />)

    await user.click(screen.getByTestId('henvisning-1963.avbryt-knapp'))

    expect(onAvbryt).toBeCalledTimes(1)
  })

  it('trykker detaljert kalkulator knapp', async () => {
    const user = userEvent.setup()
    const onAvbryt = vi.fn()
    const open = vi.fn()
    vi.stubGlobal('open', open)
    render(<Henvisning1963 onAvbryt={onAvbryt} />)

    await user.click(
      screen.getByTestId('henvisning-1963.gaa-til-detaljert-kalkulator-knapp')
    )

    expect(open).toHaveBeenCalledWith(externalUrls.detaljertKalkulator, '_self')
  })
})
