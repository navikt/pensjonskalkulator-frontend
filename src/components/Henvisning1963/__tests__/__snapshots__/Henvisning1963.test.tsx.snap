// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Henvisning1963 > rendrer 1`] = `
<DocumentFragment>
  <section
    aria-live="polite"
    class="_section_67454f _section__largePadding_67454f _section__marginBotton_67454f"
    data-testid="henvisning-1963"
  >
    <h2
      class="navds-heading navds-heading--medium navds-typo--spacing"
    >
      error.du_kan_ikke_bruke_enkel_kalkulator
    </h2>
    <div
      class="navds-stack navds-vstack"
      style="--__ac-stack-direction: column; --__ac-stack-wrap: nowrap; --__ac-stack-gap-xs: var(--a-spacing-4);"
    >
      <p
        class="navds-body-long navds-body-long--large"
      >
        henvisning1963.body
      </p>
      <div
        class="navds-stack navds-hstack"
        style="--__ac-stack-direction: row; --__ac-stack-wrap: wrap; --__ac-stack-gap-xs: var(--a-spacing-4);"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          data-testid="henvisning-1963.gaa-til-detaljert-kalkulator-knapp"
          type="button"
        >
          <span
            class="navds-label"
          >
            henvisning1963.detaljert_kalkulator
          </span>
        </button>
        <button
          class="navds-button navds-button--secondary navds-button--medium"
          data-testid="henvisning-1963.avbryt-knapp"
          type="button"
        >
          <span
            class="navds-label"
          >
            stegvisning.avbryt
          </span>
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
